x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -248,8 +248,17 @@ static String getPutMappingIndexOrAlias(PutMappingRequest request, List<String>
             if (aliasMetaData != null) {
                 Optional<String> foundAlias = aliasMetaData.stream()
                     .map(AliasMetaData::alias)
+                    .filter(aliasName -> {
+                        List<IndexMetaData> indexMetadata = metaData.getAliasAndIndexLookup().get(aliasName).getIndices();
+                        if (indexMetadata.size() == 1) {
+                            return true;
+                        } else {
+                            IndexMetaData idxMeta = ((AliasOrIndex.Alias) metaData.getAliasAndIndexLookup().get(aliasName))
+                                    .getWriteIndex();
+                            return idxMeta != null && idxMeta.getIndex().getName().equals(concreteIndexName);
```

Comment:
I think, given "authorization on aliases" is a feature, we should stick with it and support it correctly.

If I have write access to `alias-1` (but no direct access to any indices), and that alias points to `logs-${today}` _(write index)_ and `logs-*` _(read)_, then the expectation would be that I only have write access to today's logs.
The security admin would be very surprised if this allowed a user to update mappings on old (in theory, read only) log indices.

Does that answer your question?

Reply:
You & I seem to be thinking about this from different angles.
From my point of view all of this is an intention to make automatic dynamic mapping changes work when ingesting to an alias (even under a `is_write_index` scenario).
So if you write to `alias1` and that triggers a mapping update to `index1` it should work.
As a side-effect, if you do an explicit mapping update to `index1` it will also work because we can't distinguish between those cases, but we don't care because there was an avenue by which this would work, so this side-effect isn't opening up anything you couldn't already do.

You seem to be looking at this from a point of view of if I try to do a mapping update to `logs-${today}` it will work, but if I do it to `logs-${yesterday}` it will fail, even though I have the same privileges on those 2 indices.
And the reason is that the `logs` alias is writing to `logs-${today}` so that's why this mapping update is allowed.
I think it's a valid concern - security should aim to be clear and understandable, and it's not nice to have things that work/fail for reasons that are hard to discover.

But, while I appreciate the concern, I don't think this is really any different than the previous situation (before `is_write_index` existed) where you'd have separate `logs-write` and `logs-read` aliases (the former pointing to `logs-${today}`, the latter to `logs-*`).
It's a potential issue, but not one I've seen cause problems for users, and not one we need to solve here.